mcstatus/motd.py

```diff
@@ -0,0 +1,549 @@
+
+
+@dataclass
+class Motd:
+    """Represents parsed MOTD.
+
+    Basing on this class, you can write your own MOTD-to-something parser, see ``parsed`` field.
+    """
+
+    parsed: list[ParsedMotdItem]
+    """Parsed MOTD, which then will be transformed.
+
+    Bases on this field, you can easily write your own MOTD-to-something parser.
+    """
+    bedrock: bool = False
+    """Is server Bedrock Edition? Some details may change in work of this class."""
+
+    @classmethod
+    def parse(cls, raw: Union[dict, list, str], *, bedrock: bool = False) -> Self:
+        """Parse a raw MOTD to less raw MOTD (``Motd.parsed`` field).
+
+        :param raw: Raw MOTD, directly from server.
+        :param bedrock: Is server Bedrock Edition? Nothing changes here, just sets attribute.
+        :returns: New ``MOTD`` instance.
+        """
+        if isinstance(raw, str):
+            return cls(cls._parse_as_str(raw, bedrock=bedrock), bedrock)
+
+        if isinstance(raw, list):
+            raw = {"extra": raw}
+
+        return cls(cls._parse_as_dict(raw), bedrock)
+
+    @classmethod
+    def _parse_as_dict(cls, item: dict, *, auto_add: Optional[list[ParsedMotdItem]] = None) -> list[ParsedMotdItem]:
+        """Parse a MOTD when it's dict.
+
+        :param item: ``dict`` directly from the server.
+        :param auto_add: Values to add on this item.
+            Most time, this is ``Formatting`` from top level.
+        :returns: ``ParsedMotdItem`` list, which need to be passed to ``__init__``.
+        """
+        parsed_motd: list[ParsedMotdItem] = auto_add if auto_add is not None else []
+
+        if item.get("color"):
+            try:
+                parsed_motd.append(MinecraftColor[item["color"].upper()])
+            except KeyError:
+                if item["color"] == "reset":
+                    # minecraft server actually can't return {"reset": True}, instead
+                    # it will return {"color": "reset"}. but for follow naming, we
+                    # will not move `Formatting.RESET` to `MinecraftColor`.
+                    #
+                    # see https://wiki.vg/Chat#Shared_between_all_components, `color` field
+                    parsed_motd.append(Formatting.RESET)
+                else:
+                    parsed_motd.append(WebColor.parse(hex=item["color"]))
+
+        for style_key, style_val in Formatting.__members__.items():
+            if item.get(style_key.lower()) is False or item.get(style_key.lower()) == "false":
+                parsed_motd.remove(style_val)
+            elif item.get(style_key.lower()) is not None:
+                parsed_motd.append(style_val)
+
+        text = item.get("text")
+        if text is not None:
+            parsed_motd.append(text)
+        translate = item.get("translate")
+        if text is None and translate is not None:
+            parsed_motd.append(TranslateString(translate))
+        parsed_motd.append(Formatting.RESET)
```

Comment:
While I'm generally against over-use of the warlus (`:=`) operator, in this case it can simplify things a lot, and can be justfied.
```suggestion
        if (text := item.get("text")) is not None:
            parsed_motd.append(text)
        elif (translate := item.get("translate")) is not None:
            parsed_motd.append(TranslateString(translate))
        parsed_motd.append(Formatting.RESET)
```

Reply:
I did this firstly, but then found that `:=` is not supported in Python 3.7.

Reply:
I see, you could add a comment about this so that once 3.7 is dropped we'll know that this can be changed to a nicer syntax.

Reply:
Added [here](https://github.com/py-mine/mcstatus/compare/ffe1cc869956fc92fc7980d4c8a265d23adfa1b6..ae2dd4eff4d605ae236fba6a2f87e783aa51de2e#diff-c462c16c38f067b712f30e9e5d65eda0a525e97348b12492eacf676da5656ed6R261).